### PR TITLE
feat(menu): Add quickOpen option. 

### PR DESCRIPTION
**Pull request by williamernest**
Fixes: #1702 
Disables open and closing animations for the mdc-menu element.

### CODE REVIEW

File: packages/mdc-menu/foundation.js [hunk 1]
```diff
@@ -208,6 +210,11 @@ class MDCMenuFoundation extends MDCFoundation {
     this.setSelectedIndex(-1);
   }
 
+  /** @param {boolean} quickOpen */
+  quickOpen(quickOpen) {
```
Comment (kfranqueiro) [inline packages/mdc-menu/foundation.js]: Should this be called `setQuickOpen`? The current name in the context of a method makes it sound like it's actually going to open the menu.

File: packages/mdc-menu/foundation.js [hunk 2]
```diff
@@ -555,7 +562,10 @@ class MDCMenuFoundation extends MDCFoundation {
    */
   open({focusIndex = null} = {}) {
     this.adapter_.saveFocus();
-    this.adapter_.addClass(MDCMenuFoundation.cssClasses.ANIMATING_OPEN);
+
+    if (!this.quickOpen_) {
+      this.adapter_.addClass(MDCMenuFoundation.cssClasses.ANIMATING_OPEN);
```
Comment (kfranqueiro) [inline packages/mdc-menu/foundation.js]: We should probably also conditionalize the `setTimeout` code below (and same for within `close`), although it's effectively a no-op in this case right now.

File: test/unit/mdc-menu/simple.foundation.test.js [hunk 3]
```diff
@@ -116,7 +116,14 @@ test('#init throws error when the necessary DOM is not present', () => {
 testFoundation('#open adds the animation class to start an animation',
   ({foundation, mockAdapter}) => {
     foundation.open();
-    td.verify(mockAdapter.addClass(cssClasses.ANIMATING_OPEN));
+    td.verify(mockAdapter.addClass(cssClasses.ANIMATING_OPEN), {times: 1});
+  });
+
+testFoundation('#open does not add the animation class to start an animation when quickOpen is false',
```
Comment (kfranqueiro) [inline test/unit/mdc-menu/simple.foundation.test.js]: Missing a complementary test for `#close`? (Also test both `addClass` and `removeClass` after addressing my comment above.)

File: packages/mdc-menu/README.md [hunk 4]
```diff
@@ -150,6 +153,7 @@ Property | Value Type | Description
 `open` | Boolean | Proxies to the foundation's `isOpen`/(`open`, `close`) methods. 
 `items` | Array<Element> | Proxies to the foundation's container to query for all `.mdc-list-item[role]` elements.
 `itemsContainer` | Element | Queries the foundation's root element for the `mdc-menu__items` container element.
+`quickOpen` | Boolean | Proxies to the foundation's `quickOpen()` method.
```
Comment (kfranqueiro) [inline packages/mdc-menu/README.md]: Description needs updating to `setQuickOpen()` now

Comment (kfranqueiro) [reply packages/mdc-menu/README.md]: It also occurs to me that `setQuickOpen` needs to be documented under the Foundation API... and make sure to mention that it affects both open _and_ close animations. (Which makes me think maybe this isn't the best name, but not sure what is... maybe flip it to `enableAnimations`, but maybe that's too vague?)

File: packages/mdc-menu/README.md [hunk 5]
```diff
@@ -201,6 +205,7 @@ Method Signature | Description
 `open({focusIndex: ?number}) => void` | Opens the menu. Optionally accepts an object with a `focusIndex` parameter to indicate which list item should receive focus when the menu is opened. 
 `close(evt: ?Event)` | Closes the menu. Optionally accepts the event to check if the target is disabled before closing the menu.  
 `isOpen() => boolean` | Returns a boolean indicating whether the menu is open. 
+`setQuickOpen(quickOpen: boolean) => void` | Sets the menu to open and close without animation when the `open/close` methods are called.
```
Comment (kfranqueiro) [inline packages/mdc-menu/README.md]: Nit #1: "Whether the menu should open and close ..."
Nit #2: `open`/`close` (separate backticks around each word, leave the slash outside them)

Comment (williamernest) [reply packages/mdc-menu/README.md]: I changed it to "Sets whether the menu should open and close..." based on every other "set" method description in this readme starting with "Sets".

**Pull request merged**